tools/reccmp/reccmp.py

```diff
@@ -305,23 +319,53 @@ def parse_asm(file, addr, size):
           else:
             ratio = 0
 
-          print('  %s (%s / %s) is %.2f%% similar to the original' % (recinfo.name, hex(addr), hex(recinfo.addr), ratio * 100))
+          percenttext = "%.2f%%" % (ratio * 100)
+          if not plain:
+            if ratio == 1.0:
+              percenttext = colorama.Fore.GREEN + percenttext + colorama.Style.RESET_ALL
+            elif ratio > 0.8:
+              percenttext = colorama.Fore.YELLOW + percenttext + colorama.Style.RESET_ALL
+            else:
+              percenttext = colorama.Fore.RED + percenttext + colorama.Style.RESET_ALL
+
+          if not verbose:
+            print('  %s (%s / %s) is %s similar to the original' % (recinfo.name, hex(addr), hex(recinfo.addr), percenttext))
 
           function_count += 1
           total_accuracy += ratio
 
           if recinfo.size:
-            if verbose == addr or html:
-              udiff = difflib.unified_diff(origasm, recompasm)
+            udiff = difflib.unified_diff(origasm, recompasm, n=10)
 
-              if verbose == addr:
+            # If verbose, print the diff for that funciton to the output
+            if verbose:
+              if ratio == 1.0:
+                print("%s: %s 100%% match.\n\nOK!" % (hex(addr), recinfo.name))
+              else:
                 for line in udiff:
-                  print(line)
-                print()
-                print()
-
-              if html:
-                htmlinsert.append('{address: "%s", name: "%s", matching: %s, diff: "%s"}' % (hex(addr), recinfo.name, str(ratio), '\\n'.join(udiff).replace('"', '\\"').replace('\n', '\\n')))
+                  if line.startswith("++") or line.startswith("@@") or line.startswith("--"):
+                    # Skip unneeded parts of the diff for the brief view
+                    pass
+                  elif line.startswith("+"):
+                    if plain:
+                      print(line)
+                    else:
+                      print(colorama.Fore.GREEN + line)
+                  elif line.startswith("-"):
+                    if plain:
+                      print(line)
+                    else:
+                      print(colorama.Fore.RED + line)
+                  else:
+                    print(line)
+                  if not plain:
+                    print(colorama.Style.RESET_ALL, end='')
+
+                print("\n%s is only %s similar to the original, diff above" % (recinfo.name, percenttext))
+
+            # If html, record the diffs to an HTML file
+            if html:
+              htmlinsert.append('{address: "%s", name: "%s", matching: %s, diff: "%s"}' % (hex(addr), recinfo.name, str(ratio), '\\n'.join(udiff).replace('"', '\\"').replace('\n', '\\n')))
```

Comment:
Looks like you're creating json with strings.
How about using python dicts and lists to gather all data,
and at the end use [`the json module`](https://docs.python.org/3/library/json.html) to create a json string.

Reply:
This is an existing line, just shuffled around, not going to go cleaning up the whole script at the moment.

Reply:
Let's hope matt reads this ;)

Reply:
I guess I never thought of it as JSON because it's just inserting data into the HTML template, but technically yeah I guess it's a JavaScript object, so it's JSON.